client/getversions.go

```diff
@@ -26,29 +26,29 @@ For more about knox, see https://github.com/pinterest/knox.
 See also: knox keys, knox get
 	`,
 }
-var getVersionsState = cmdGetVersions.Flag.String("s", "ACTIVE", "")
+var getVersionsState = cmdGetVersions.Flag.String("s", "active", "")
 
 func runGetVersions(cmd *Command, args []string) {
 	if len(args) != 1 {
 		fatalf("get takes only one argument. See 'knox help versions'")
 	}
 
-	keyID := args[0]
-	key, err := cli.GetKey(keyID)
-	if err != nil {
-		fatalf("Error getting key: %s", err.Error())
-	}
-	var kvl knox.KeyVersionList
+	var status knox.VersionStatus
 	switch strings.ToLower(*getVersionsState) {
 	case "active":
-		kvl = key.VersionList.GetActive()
-	case "inactive":
-		kvl = key.VersionList
+		status = knox.Active
 	case "primary":
-		kvl = knox.KeyVersionList{*key.VersionList.GetPrimary()}
-	default:
-		fatalf("Invalid status parameter: %s", *getVersionsState)
+		status = knox.Primary
+	case "inactive":
+		status = knox.Inactive
+	}
+
+	keyID := args[0]
+	key, err := cli.GetKeyWithStatus(keyID, status)
+	if err != nil {
+		fatalf("Error getting key: %s", err.Error())
 	}
+	kvl := key.VersionList
 	for _, v := range kvl {
 		fmt.Printf("%d\n", v.ID)
```

Comment:
can you add the status to the display? been meaning to do that.

Reply:
We're sure no automation is using this? If so, can definitely add that.

Reply:
maybe lets add a verbose flag just in case.